ee/server/configuration/saml.ts

```diff
@@ -38,6 +40,29 @@ onLicense('saml-enterprise', () => {
 			metadataTemplate: settings.get(`${ service }_Metadata_template`),
 		});
 	});
+
+	SAMLUtils.events.on('updateCustomFields', (loginResult: Record<string, any>, updatedUser: {userId: string; token: string}) => {
+		const userDataCustomFieldMap = settings.get('SAML_Custom_Default_user_data_custom_fieldmap') as string;
+		const customMap: Record<string, any> = JSON.parse(userDataCustomFieldMap);
+
+		const customFieldsList: Record<string, any> = {};
+
+		for (const spCustomFieldName in customMap) {
+			if (!customMap.hasOwnProperty(spCustomFieldName)) {
+				continue;
+			}
+
+			const customAttribute = customMap[spCustomFieldName];
+			const value = SAMLUtils.getProfileValue(loginResult.profile, { fieldName: spCustomFieldName });
+			customFieldsList[customAttribute] = value;
+		}
+
+		Users.update(updatedUser.userId, {
```

Comment:
please move this update to a model action `updateCustomFieldsById(userId, customFieldsList)`